cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Load.java

```diff
@@ -95,6 +96,40 @@ public double expectedUtilizationFor(Resource resource) {
     return expectedUtilizationFor(resource, _metricValues, false);
   }
 
+  /**
+   * Get a single snapshot value that is representative for the given KafkaMetric type. The current algorithm uses
+   * <ol>
+   *   <li>If the max load is not requested, it is max/latest/mean load depending on the ValueComputingStrategy
+   *   which the KafkaMetric type uses.</li>
+   *   <li>If the max load is requested, it is the max load.</li>
+   * </ol>
+   *
+   * @param metric KafkaMetric type for which the expected utilization will be provided.
+   * @param wantMaxLoad True if the requested utilization represents the peak load, false otherwise.
+   * @return A single representative utilization value on a metric type.
+   */
+  public double expectedUtilizationFor(KafkaMetricDef metric, boolean wantMaxLoad) {
+    MetricInfo info;
+    switch (metric.defScope()) {
+      case COMMON:
+        info = KafkaMetricDef.commonMetricDef().metricInfo(metric.name());
+        break;
+      case BROKER_ONLY:
+        info = KafkaMetricDef.brokerMetricDef().metricInfo(metric.name());
+        break;
+      default:
+        throw new IllegalArgumentException("Metric scope " + metric.defScope() + " for metric " + metric.name() + " is invalid.");
+    }
+    if (_metricValues.isEmpty()) {
+      return 0.0;
+    }
+    double result = 0;
+    MetricValues valuesForId = _metricValues.valuesFor(info.id());
+    result += (wantMaxLoad || metric.valueComputingStrategy() == ValueComputingStrategy.MAX) ? valuesForId.max() :
```

Comment:
Would using `switch` - `case` here help us:
(1) check to ensure that `ValueComputingStrategy` is either `MAX`, `LATEST`, or `AVG`, handle them accordingly, and throw `IllegalArgumentException` otherwise,
(2) make the ternary logic more clear.